prediction_market_agent/agents/microchain_agent/microchain_generators.py

```diff
@@ -0,0 +1,125 @@
+import typing as t
```

Comment:
I'd like to get this reviewed here and merge it in. And then add it to the microchain itself and remove it here.